### PR TITLE
New Package: EcoSLIM

### PR DESCRIPTION
**Pull request by smithsg84**
Requesting EcoSLIM particle code package addition.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ecoslim/package.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    version('1.3', sha256='b532e570b4767e4fa84123d8773732150679e8e3d7fecd5c6e99fb1d4dc57b84')
+    version('develop', branch='develop')
+
+    def cmake_args(self):
+        """Populate cmake arguments for EcoSLIM."""
+        spec = self.spec
+
+        cmake_args = []
+
+        return cmake_args
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ecoslim/package.py]: Are you sure you don't want/need any `cmake_args`?  If so, change these lines to the following:

```suggestion
        return []
```

File: var/spack/repos/builtin/packages/ecoslim/package.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+from spack import *
+
+class Ecoslim(CMakePackage):
+    """EcoSLIM is a Lagrangian, particle-tracking code that simulates
+    advective and diffusive movement of water parcels.  EcoSLIM
+    integrates seamlessly with ParFlow-CLM."""
+
+    homepage = "https://github.com/reedmaxwell/EcoSLIM"
+    url      = "https://github.com/reedmaxwell/EcoSLIM/archive/refs/tags/v1.3.tar.gz"
+    git      = "git@github.com:reedmaxwell/EcoSLIM.git"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ecoslim/package.py]: For consistency with nearly all other Spack packages:

```suggestion
    git      = "https://github.com/reedmaxwell/EcoSLIM.git"
```

File: var/spack/repos/builtin/packages/ecoslim/package.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    maintainers = ['reedmaxwell', 'lecondon', 'smithsg84']
+
+    version('1.3', sha256='b532e570b4767e4fa84123d8773732150679e8e3d7fecd5c6e99fb1d4dc57b84')
+    version('develop', branch='develop')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ecoslim/package.py]: The software doesn't have a `develop` branch.  Do you mean `master`?

```suggestion
    version('master', branch='master')
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
A few questions/comments based on current contents (and assuming you didn't mean to make this a `draft` PR :grin:).

**Comment by tldahlgren** [comment]
I also confirmed the sha256 for v1.3 (using `spack checksum`).

File: var/spack/repos/builtin/packages/ecoslim/package.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+class Ecoslim(CMakePackage):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ecoslim/package.py:9]:
```suggestion

class Ecoslim(CMakePackage):
```

File: var/spack/repos/builtin/packages/ecoslim/package.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+    maintainers = ['reedmaxwell', 'lecondon', 'smithsg84']
+
+    version('1.3', sha256='b532e570b4767e4fa84123d8773732150679e8e3d7fecd5c6e99fb1d4dc57b84')
+    version('master', branch='master')
+
+    def cmake_args(self):
+        """Populate cmake arguments for EcoSLIM."""
+        return []
+
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ecoslim/package.py]:
```suggestion
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]

**Comment by tldahlgren** [comment]
@smithsg84 Hope you don't mind that  I committed the two simple fixes reported by the style checker.

**Comment by smithsg84** [comment]
> @smithsg84 Hope you don't mind that I committed the two simple fixes reported by the style checker.

Sorry I forgot about the spack style command.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**